src/middlewares/debugMiddleware.ts

```diff
@@ -0,0 +1,79 @@
+import { CONFIG } from '../config'
+import * as crypto from '@shardus/crypto-utils'
+import { nestedCountersInstance } from '../utils/nestedCounters'
+
+const MAX_COUNTER_BUFFER_MILLISECONDS = 10000
+let lastCounter = 0
+let multiSigLstCounter = 0
+
+/** Helper Functions */
+
+function verify(obj: crypto.SignedObject, expectedPk?: string): boolean {
+    try {
+      if (expectedPk) {
+        if (obj.sign.owner !== expectedPk) return false
+      }
+      return crypto.verifyObj(obj)
+    } catch (e) {
+      console.error(`Error in verifying object ${JSON.stringify(obj)}`, e)
+      return false
+    }
+}
+
+export function ensureKeySecurity(pubKey: string, level: number): boolean {
+    const devPublicKeys = getDevPublicKeys()
+    // eslint-disable-next-line security/detect-object-injection
+    const pkClearance = devPublicKeys[pubKey]
+    return pkClearance !== undefined && pkClearance >= level
+}
+
+export function getDevPublicKeys(): {[pubkey:string]: number}  {
+    return CONFIG.devPublicKeys || {};
+}
+
+// This function is used to check if the request is authorized to access the debug endpoint
+export function handleDebugAuth(_req: any, res: any, next: any, authLevel: any) 
```

Comment:
This middleware or the endpoints that call it need a rate limiter. See service points in the validator repo for example.